assets/app/scripts/extensions/javaLink.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsoleExtensions.extension:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
+  .factory("ProxyPod", function(API_CFG) {
+    return function(namespace, podId, port) {
+      var kubeProxyURI = new URI(window.location.protocol + '//' + API_CFG.k8s.hostPort + API_CFG.k8s.prefix);
+      var apiVersion = API_CFG.k8s.version || 'v1beta3';
+      if (port) {
+        podId = podId + ':' + port;
+      }
+      kubeProxyURI.path(UrlHelpers.join(kubeProxyURI.path(), 'v1beta3/proxy/namespaces', namespace, 'pods', podId));
+      return kubeProxyURI.toString();
+    };
+  })
+  .run(function(ProxyPod, BaseHref, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="container.jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView(container, container.jolokiaUrl)" href="">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+      var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
```

Comment:
is this the only thing being used from the imported hawtio-utilities? If so, can you just inline the function? I'd like to eventually find a way to render the link without traversing angular scopes hunting for a pod object anyway, at which point we wouldn't need `findParentWith`